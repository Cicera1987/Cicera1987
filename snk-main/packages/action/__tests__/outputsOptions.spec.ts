import { parseEntry } from "../outputsOptions";

[
  "path/to/out.gif",

  "/out.svg?color_snake=orange&color_dots=#000,#111,#222,#333,#444",

  `/out.svg?{"color_snake":"yellow","color_dots":["#000","#111","#222","#333","#444"]}`,

  `/out.svg       {"color_snake":"yellow"}`,

  "/out.svg?color_snake=orange&color_dots=#000,#111,#222,#333,#444&dark_color_dots=#a00,#a11,#a22,#a33,#a44",

  "/out.svg?.gif.svg?color_snake=orange",
].forEach((entry) =>
  it(`should parse ${entry}`, () => {
    expect(parseEntry(entry)).toMatchSnapshot();
  })
);
