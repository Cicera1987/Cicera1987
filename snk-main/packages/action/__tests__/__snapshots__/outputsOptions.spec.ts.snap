// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should parse /out.svg       {"color_snake":"yellow"} 1`] = `
Object {
  "animationOptions": Object {
    "frameDuration": 100,
    "step": 1,
  },
  "drawOptions": Object {
    "colorDotBorder": "#1b1f230a",
    "colorDots": Array [
      "#ebedf0",
      "#9be9a8",
      "#40c463",
      "#30a14e",
      "#216e39",
    ],
    "colorEmpty": "#ebedf0",
    "colorSnake": "yellow",
    "dark": Object {
      "colorDotBorder": "#1b1f230a",
      "colorDots": Array [
        "#161b22",
        "#01311f",
        "#034525",
        "#0f6d31",
        "#00c647",
      ],
      "colorEmpty": "#161b22",
      "colorSnake": "purple",
    },
    "sizeCell": 16,
    "sizeDot": 12,
    "sizeDotBorderRadius": 2,
  },
  "filename": "/out.svg",
  "format": "svg",
}
`;

exports[`should parse /out.svg?.gif.svg?color_snake=orange 1`] = `
Object {
  "animationOptions": Object {
    "frameDuration": 100,
    "step": 1,
  },
  "drawOptions": Object {
    "colorDotBorder": "#1b1f230a",
    "colorDots": Array [
      "#ebedf0",
      "#9be9a8",
      "#40c463",
      "#30a14e",
      "#216e39",
    ],
    "colorEmpty": "#ebedf0",
    "colorSnake": "orange",
    "dark": Object {
      "colorDotBorder": "#1b1f230a",
      "colorDots": Array [
        "#161b22",
        "#01311f",
        "#034525",
        "#0f6d31",
        "#00c647",
      ],
      "colorEmpty": "#161b22",
      "colorSnake": "purple",
    },
    "sizeCell": 16,
    "sizeDot": 12,
    "sizeDotBorderRadius": 2,
  },
  "filename": "/out.svg?.gif.svg",
  "format": "svg",
}
`;

exports[`should parse /out.svg?{"color_snake":"yellow","color_dots":["#000","#111","#222","#333","#444"]} 1`] = `
Object {
  "animationOptions": Object {
    "frameDuration": 100,
    "step": 1,
  },
  "drawOptions": Object {
    "colorDotBorder": "#1b1f230a",
    "colorDots": Array [
      "#000",
      "#111",
      "#222",
      "#333",
      "#444",
    ],
    "colorEmpty": "#000",
    "colorSnake": "yellow",
    "dark": undefined,
    "sizeCell": 16,
    "sizeDot": 12,
    "sizeDotBorderRadius": 2,
  },
  "filename": "/out.svg",
  "format": "svg",
}
`;

exports[`should parse /out.svg?color_snake=orange&color_dots=#000,#111,#222,#333,#444 1`] = `
Object {
  "animationOptions": Object {
    "frameDuration": 100,
    "step": 1,
  },
  "drawOptions": Object {
    "colorDotBorder": "#1b1f230a",
    "colorDots": Array [
      "#000",
      "#111",
      "#222",
      "#333",
      "#444",
    ],
    "colorEmpty": "#000",
    "colorSnake": "orange",
    "dark": undefined,
    "sizeCell": 16,
    "sizeDot": 12,
    "sizeDotBorderRadius": 2,
  },
  "filename": "/out.svg",
  "format": "svg",
}
`;

exports[`should parse /out.svg?color_snake=orange&color_dots=#000,#111,#222,#333,#444&dark_color_dots=#a00,#a11,#a22,#a33,#a44 1`] = `
Object {
  "animationOptions": Object {
    "frameDuration": 100,
    "step": 1,
  },
  "drawOptions": Object {
    "colorDotBorder": "#1b1f230a",
    "colorDots": Array [
      "#000",
      "#111",
      "#222",
      "#333",
      "#444",
    ],
    "colorEmpty": "#000",
    "colorSnake": "orange",
    "dark": Object {
      "colorDots": Array [
        "#a00",
        "#a11",
        "#a22",
        "#a33",
        "#a44",
      ],
      "colorEmpty": "#a00",
    },
    "sizeCell": 16,
    "sizeDot": 12,
    "sizeDotBorderRadius": 2,
  },
  "filename": "/out.svg",
  "format": "svg",
}
`;

exports[`should parse path/to/out.gif 1`] = `
Object {
  "animationOptions": Object {
    "frameDuration": 100,
    "step": 1,
  },
  "drawOptions": Object {
    "colorDotBorder": "#1b1f230a",
    "colorDots": Array [
      "#ebedf0",
      "#9be9a8",
      "#40c463",
      "#30a14e",
      "#216e39",
    ],
    "colorEmpty": "#ebedf0",
    "colorSnake": "purple",
    "dark": Object {
      "colorDotBorder": "#1b1f230a",
      "colorDots": Array [
        "#161b22",
        "#01311f",
        "#034525",
        "#0f6d31",
        "#00c647",
      ],
      "colorEmpty": "#161b22",
      "colorSnake": "purple",
    },
    "sizeCell": 16,
    "sizeDot": 12,
    "sizeDotBorderRadius": 2,
  },
  "filename": "path/to/out.gif",
  "format": "gif",
}
`;
